microcosm_flask/audit.py

```diff
@@ -45,6 +46,33 @@ def should_skip_logging(func):
     return getattr(func, SKIP_LOGGING, False)
 
 
+@contextmanager
+def logging_levels():
+    """
+    Context manager to conditionally set logging levels.
+
+    Supports setting pre-request, comma-delimited logging levels using the `X-Debug` header.
+    Maps the root logger to "*" for disambiguation.
+
+    """
+    # extra logger names from headers
+    loggers = [
+        "" if logger.strip() == "*" else logger.strip()
+        for logger in request.headers.get("x-debug", "").split(",")
```

Comment:
I wonder if we could leverage `graph.opaque` instead of explicitly calling `request.headers` and maybe put all this in `microcosm-logging` (which already depends on microcosm)?

I understand we should try it out first and abstract later but just a thought that occurred to me after thinking we'll want the same thing for the pubsub layer, especially because more of our debug logs will be there.